clearnode/contract_event.go

```diff
@@ -63,3 +63,18 @@ func GetLatestContractEvent(db *gorm.DB, contractAddress string, networkID uint3
 
 	return &ev, err
 }
+
+func IsContractEventPresent(db *gorm.DB, chainID uint32, txHash string, logIndex uint32) (bool, error) {
```

Comment:
You also need to pass contract address here

Reply:
Is there really a chance we will have multiple events per chainID, txHash, logIndex?)

Reply:
> Is there really a chance we will have multiple events per chainID, txHash, logIndex?)

No, it is impossible. Moreover, `chainID` is redundant as well, because it is used to calculate `txHash`.